src/ZeldaWindWaker/m_do_ext.ts

```diff
@@ -85,6 +101,221 @@ export class mDoExt_bvaAnm extends mDoExt_baseAnm<VAF1> {
+        colorCopy(materialParams.u_Color[ColorKind.C1], this.tevStr.colorK0);
+        colorCopy(materialParams.u_Color[ColorKind.C2], this.color);
+        mat4.copy(drawParams.u_PosMtx[0], globals.camera.viewFromWorldMatrix);
+
+        this.ddraw.beginDraw(globals.modelCache.cache);
+        for (let i = 0; i < this.numLines; i++) {
+            const line = this.lines[i];
+            this.ddraw.begin(GX.Command.DRAW_TRIANGLE_STRIP, this.numSegments * 2);
+            for (let j = 0; j < this.numSegments * 2; j += 2) {
+                this.ddraw.position3vec3(line.positions[j + 0]);
+                this.ddraw.texCoord2vec2(GX.Attr.TEX0, line.texCoords![j + 0]);
+                this.ddraw.normal3f32(0.25, 0.0, 0.0);
+
+                this.ddraw.position3vec3(line.positions[j + 1]);
+                this.ddraw.texCoord2vec2(GX.Attr.TEX0, line.texCoords![j + 1]);
+                this.ddraw.normal3f32(-0.25, 0.0, 0.0);
+            }
+            this.ddraw.end();
+        }
+        this.ddraw.endDraw(renderInstManager);
+
+        const renderInst = this.ddraw.makeRenderInst(renderInstManager);
+        renderInst.setSamplerBindingsFromTextureMappings(materialParams.m_TextureMapping);
+        this.material.allocateMaterialParamsDataOnInst(renderInst, materialParams);
+        this.material.allocateDrawParamsDataOnInst(renderInst, drawParams);
+        this.material.setOnRenderInst(renderInstManager.gfxRenderCache, renderInst);
+
+        renderInstManager.submitRenderInst(renderInst);
+    }
+
+    public updateWithScale(globals: dGlobals, segmentCount: number, scale: number, color: Color, space: number, tevStr: dKy_tevstr_c): void {
+        this.color = color;
+        this.tevStr = tevStr;
+        this.numSegments = Math.min(segmentCount, this.maxSegments);
+        const spacing = (space !== 0) ? scale / space : 0.0;
+
+        let dist = 0.0;
+        for (let i = 0; i < this.numLines; i++) {
+            const line = this.lines[i];
+            const srcPos = line.segments;
+            let segIdx = 0;
+            let vertIdx = 0;
+            assert(!!line.texCoords);
+
+            const dstPos = line.positions;
+            const dstUvs = line.texCoords;
+            let r_scale = scale;
+
+            // Handle the first segment
+            dstUvs[vertIdx + 0][1] = dist;
+            dstUvs[vertIdx + 1][1] = dist;
+
+            const segVec = vec3.sub(scratchVec3a, srcPos[segIdx + 1], srcPos[segIdx + 0]);
+            const delta = vec3.length(segVec);
+            dist += delta * 0.1;
+
+            // Normalize and then scale a vector orthogonal to both the segment and eye
+            const eyeVec = vec3.sub(scratchVec3b, srcPos[segIdx + 0], globals.camera.cameraPos);
+            const eyeCross = vec3.cross(scratchVec3a, segVec, eyeVec);
+            const mag = vec3.length(eyeCross);
+            if (mag !== 0.0) {
+                vec3.scale(eyeCross, eyeCross, scale / mag);
+            }
```

Comment:
```suggestion
            normalizeToLength(eyeCross, eyeCross, scale);
```

(I now realize it should be called "width" instead of "scale")